serde/src/de/impls.rs

```diff
@@ -91,6 +151,29 @@ impl Visitor for BoolVisitor {
             _ => Err(Error::invalid_type(Type::Bool)),
         }
     }
+
+    #[cfg(all(not(feature = "std"), feature = "rustc_unicode"))]
+    fn visit_str<E>(&mut self, s: &str) -> Result<bool, E>
```

Comment:
This doesn't seem worthwhile if we need to bundle in our own trim implementation.